hazelcast/pom.xml

```diff
@@ -345,6 +345,39 @@
                             </target>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>copy-and-fill-hazelcast-assembly-xml</id>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <target>
+                                <copy todir="${basedir}/target/classes">
+                                    <fileset dir="${basedir}/src/main/config-template" casesensitive="yes">
+                                        <include name="hazelcast-assembly.*"/>
+                                    </fileset>
+                                    <globmapper from="hazelcast-assembly.*" to="hazelcast-default-assembly.*"/>
+                                    <filterset>
+                                        <filter token="FILTER_LOOPBACK_INTERFACE" value="true" />
+                                        <filter token="FILTER_BIND_ANY" value="false" />
+                                        <filter token="FILTER_DISTRO_NAME" value="ZIP/TGZ" />
+                                    </filterset>
+                                </copy>
+                                <copy todir="${basedir}/target/classes">
+                                    <fileset dir="${basedir}/src/main/config-template" casesensitive="yes">
+                                        <include name="hazelcast-assembly.*"/>
+                                    </fileset>
+                                    <globmapper from="hazelcast-assembly.*" to="hazelcast-docker-assembly.*"/>
+                                    <filterset>
+                                        <filter token="FILTER_LOOPBACK_INTERFACE" value="false" />
+                                        <filter token="FILTER_BIND_ANY" value="true" />
+                                        <filter token="FILTER_DISTRO_NAME" value="Docker" />
+                                    </filterset>
+                                </copy>
+                            </target>
+                        </configuration>
+                    </execution>
```

Comment:
Not sure if this is better than having yet another copy of the file. Yes, updates will be easier, but if I want to have a look at what the particular file actually contains (and what files are generated) I need to look here, instead of the file.

From the other POV there is also the `hazelcast-default.xml`, which is also almost identical, so if we are generating 2, we could generate 3.

Reply:
@mmedenjak wdyt?

Reply:
Josef pinged me, so - I guess it’s fine. I can say it was surprising at first, but I can see the benefits and it just needs some getting used to. If we start getting more and more differences between the configs, then I’d have separate copies.
But it’s not a strong opinion, I think we’ll live with either.

Reply:
I say - let's try with the template and if we start getting many questions on slack about how it's supposed to work, then we can easily send a PR with the copies. I don't think it's a big deal, whichever way we choose.